### ISSUE TITLE
Autocomplete does not handle column names with international characters correctly

### ISSUE THREAD

**Issue by stegus64** [dbeaver/dbeaver#21650]
### Description

We have tables in a databricks database that have swedish column names. One example is the swedish word Resultatställe. Swedish words often contains the characters åäö ÅÄÖ.

DBeaver lists these columns correctly in the autocomplete dropdown:

<img width="127" alt="image" src="https://github.com/dbeaver/dbeaver/assets/51655419/eca0d7f7-d8ff-4167-94e8-54eb3f59eeac">

But when selecting a column the generated code causes an error in Databricks:

<img width="228" alt="image" src="https://github.com/dbeaver/dbeaver/assets/51655419/bc0cc398-14d8-4d59-9097-4d515e77eb85">

To make the code work we need to manually add backquoutes around the name:

<img width="200" alt="image" src="https://github.com/dbeaver/dbeaver/assets/51655419/b44ce7e9-8fd0-4192-aae6-7ed48d7d943c">

It would be much better if DBeaver automatically added these backticks if a column name contains any non-ascii characters.

The problem also affects the * expansion. The generated list of columns should have backticks around each column name with non-ascii characters.

### DBeaver Version

Community edition Version 23.2.3.202310221533

### Operating System

Windows 11

### Database and driver

Azure Databricks
Downloads\SimbaSparkJDBC42-2.6.22.1040\SparkJDBC42.jar

### Steps to reproduce

See description

**Comment by E1izabeth** [comment]
Thank you for bug report. I believe it should be fixed in scope of #21589

**Comment by mkgrgis** [comment]
Similar problem is if there is white space character inside of name - no quoting since some of last versions, but there was.

**Comment by E1izabeth** [comment]
@mkgrgis I can't reproduce it on 23.2.4. Please check that you're on the latest DBeaver version and specify your database if it's still reproducible for you.

**Comment by E1izabeth** [comment]
@stegus64 does it still actual for the latest DBeaver version?

**Comment by stegus64** [comment]
> @stegus64 does it still actual for the latest DBeaver version?

Yes, I just installed 23.2.4 and I still have exactly the same problem.

**Comment by E1izabeth** [comment]
Thank you for update

**Comment by stegus64** [comment]
I made some more test. When connected to SQL Server, everything works as expected with both column names with spaces and column names with non-ascii characters. 

But, it does not work with Databricks and delta tables. I do not think that spaces in column names are supported at all in delta tables, but non-ascii characters are supported, but you must use backticks for quoting. DBeaver does not generate these backticks.

**Comment by mkgrgis** [comment]
Problem is also applicable for both PotgreSQL and Oracle in DBeaver 23.2.4 with greek, georgian and cyrilic names both with spaces and without. Please also test mixedcase greek and cyrilic names for representative fixing. I think autocomplete shoud detect mixedcase names or names with space and add quotes even if "Case sensitive metatdata" is off.

**Comment by ShadelessFox** [comment]
Hello @mkgrgis,

Please provide an example

**Comment by mkgrgis** [comment]
PostgreSQL `select * from "Ελληνική μυθολογία"."θεούς του Ολύμπου";`
![изображение](https://github.com/dbeaver/dbeaver/assets/41448637/ee4ad03e-4df0-438f-8f6b-32d8c8267d7c)
No autocompleteion with and without `"`
![изображение](https://github.com/dbeaver/dbeaver/assets/41448637/fc489790-c6b7-4c65-95d3-589390a7dc48)
The same case with names from scripts https://github.com/mkgrgis/Parc_de_Pavlovsk_PostGIS/tree/main/SQL ,
![изображение](https://github.com/dbeaver/dbeaver/assets/41448637/3f9514c1-6005-43ea-a434-7eb77f48cdc6)
The same with georgian letters with and without whitespaces.

**Comment by ShadelessFox** [comment]
Thank you.

**Comment by ShadelessFox** [comment]
Hello @stegus64,

May I ask you how did you manage to create a table with such a name?

The issue is that Databricks, in fact, limits table names to only contain alphabet characters, numbers, and an underscore, hence I'm unable to reproduce your issue.

Thanks.

**Comment by stegus64** [comment]
The problem in my case is not the name of the table, it is the name of table columns. This creates a table with some columns containing swedish characters:

```
-- Table
CREATE TABLE test2 (
  normalcolumn VARCHAR(100),
  `Ställe` VARCHAR(100),
  `åäöÅÄÖ` varchar(100)
  )
USING delta
;

--View
create view view2
as select * from test2
;
```

**Comment by ShadelessFox** [comment]
I completely missed that you were talking about table columns. Thanks for pointing that out.

**Comment by ShadelessFox** [comment]
Hello @mkgrgis,

I was able to reproduce your issue, but it seems like a completely different story. In the table name, the first letter is the `Ε` letter from the Greek alphabet, not the `E` letter from the English alphabet.

I assume that you were typing the one from the English alphabet.

Treating both letters equally is impossible; therefore, your issue can't be fixed.

**Comment by mkgrgis** [comment]
Hello @ShadelessFox !

> I was able to reproduce your issue, but it seems like a completely different story.

Why? I have problems with german names of tables, also with greek, georgian and cyrilic. Look like all this cases is out of mideveal latin alphabet (or so called "latin 0"). ISO:SQL never recommends to limit identifiers to latin 0, this is applicable only to some legacy or very special DBMS (maybe, to GridDB and InfluxDB I am not sure).

> In the table name, the first letter is the `Ε` letter from the Greek alphabet, not the `E` letter from the English alphabet.

Yes.

> I assume that you were typing the one from the English alphabet.

No, I have also problems with german names autocompletion but don't think it's something different with problems around of independent alphabets because @stegus64 yet have described some similar problem.

> Treating both letters equally is impossible; therefore, your issue can't be fixed.

Why? In some DBeaver versions there was normal greek and cyrilic autocompletion. Is internal DBeaver SQL AST (abstract syntax tree) waiting on ISO:SQL identifier in proper places or something latin only? Is there in Java ICU classifiers implementation? If there is both latin, greek, cyrilic and georgian letters should treated as valid identifier in SQLite, PostgreSQL, Oracle, Firebird, MariaDB, Redis and some other DBs out of my experience. How can I help you, @ShadelessFox ?

**Comment by ShadelessFox** [comment]
> Why?

The original issue is about identifiers not being quoted, and yours is about missing autocompletion proposals.

> Is internal DBeaver SQL AST (abstract syntax tree) waiting on ISO:SQL identifier in proper places or something latin only

No, it only relies on fetched metadata from system tables and filters proposals based on the input data.

> Is there in Java ICU classifiers implementation

I believe you're talking about transliteration, which was never used in DBeaver for autocompletion.

Although some characters look the same when rendered on the screen, they serve different purposes in different languages, so they shouldn't be treated "equally".

**Comment by mkgrgis** [comment]
> I believe you're talking about transliteration, which was never used in DBeaver for autocompletion.

No. I am about standard ICU classifiers "isLetter", "isDigit" (see also https://github.com/dbeaver/dbeaver/issues/18973) "isPunctuationSign" etc.

> Although some characters look the same when rendered on the screen, they serve different purposes in different languages, so they shouldn't be treated "equally".

Yes, but my message was about greek keyboard using. If I **begin without quoting but full name contains whitespace**.

**Comment by ShadelessFox** [comment]
@mkgrgis can you please create a new issue with all provided information?
